clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1559,15 +1559,24 @@ def ComplexBinOp : CIR_Op<"complex.binop",
 //===----------------------------------------------------------------------===//
 
 class CIR_BitOp<string mnemonic, TypeConstraint inputTy>
-    : CIR_Op<mnemonic, [Pure]> {
+    : CIR_Op<mnemonic, [Pure, SameOperandsAndResultType]> {
   let arguments = (ins inputTy:$input);
-  let results = (outs SInt32:$result);
+  let results = (outs inputTy:$result);
 
   let assemblyFormat = [{
     `(` $input `:` type($input) `)` `:` type($result) attr-dict
   }];
 }
 
+class CIR_CountZerosBitOp<string mnemonic, TypeConstraint inputTy>
+    : CIR_BitOp<mnemonic, inputTy> {
+  let arguments = (ins inputTy:$input, I1Attr:$is_zero_poison);
```

Comment:
Do not use `I1Attr`; use `UnitAttr` instead. Then use [optional groups](https://mlir.llvm.org/docs/DefiningDialects/Operations/#optional-groups) in the assembly format for `is_zero_poison`.

Reply:
I think we should use `I1Attr` here to be consist with MLIR https://github.com/llvm/clangir/blob/52fd4a239760a2356a0c01a5f14686ca9ad96567/mlir/include/mlir/Dialect/LLVMIR/LLVMIntrinsicOps.td#L93

Reply:
I believe CIR always uses `UnitAttr` for representing boolean flags.